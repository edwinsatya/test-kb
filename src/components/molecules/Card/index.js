import React from "react";
import ImageWrapper from "../../atoms/ImageWrapper";
import Title from "../../atoms/Title";
import ProgressBar from "../../atoms/ProgressBar";

export default function Card({ classProps, dataProps }) {
  if (dataProps) {
    return (
      <div className={classProps}>
        <ImageWrapper dataProps={dataProps} />
        <div className="flex flex-col p-1 md:p-2 xl:p-3">
          <Title
            classProps={
              "text-[10px] md:text-lg xl:text-2xl font-semibold my-1 xl:my-2 2xl:my-3"
            }
            title={
              dataProps.campaignName.length <= 13
                ? dataProps.campaignName
                : dataProps.campaignName.substr(0, 12) + "..."
            }
          />
          <ProgressBar dataProps={dataProps.percentage} />
          <div className="flex flex-col">
            <div className="flex justify-between">
              <Title
                classProps={"text-[6px] md:text-base xl:text-lg font-semibold"}
                title="Terkumpul"
              />
              <Title
                classProps={"text-[6px] md:text-base xl:text-lg font-semibold"}
                title="Sisa hari"
              />
            </div>
            <div className="flex justify-between">
              <Title
                classProps={"text-[6px] md:text-base xl:text-lg font-semibold"}
                title={dataProps.donationReceived}
              />
              <Title
                classProps={"text-[6px] md:text-base xl:text-lg font-semibold"}
                title={dataProps.remainderDay}
              />
            </div>
          </div>
        </div>
      </div>
    );
  } else {
    return <div>No Data</div>;
  }
}
