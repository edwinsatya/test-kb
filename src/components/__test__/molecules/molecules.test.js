import React from "react";
import renderer from "react-test-renderer";
import Card from "../../molecules/Card";

const dataProps = [
  {
    campaignName: "Tribute to Seventeen",
    imageUrl:
      "https://kitabisa-userupload-01.s3-ap-southeast-1.amazonaws.com/_production/campaign/97347/31_97347_1545999748_841841_f.png",
    percentage: 0,
    donationReceived: "Rp50.000",
    remainderDay: 0,
    donationGoal: 500000000,
  },
  {
    campaignName: "Tribute to Seventeen",
    imageUrl:
      "https://kitabisa-userupload-01.s3-ap-southeast-1.amazonaws.com/_production/campaign/97347/31_97347_1545999748_841841_f.png",
    percentage: 0,
    donationReceived: "Rp50.000",
    remainderDay: 0,
    donationGoal: 500000000,
  },
];

describe("Card Component", () => {
  it("Should render with no dataProps from api", () => {
    let component = renderer.create(<Card />);
    expect(component).toMatchSnapshot();
  });

  it("Should render with dataProps from api", () => {
    let component = renderer.create(<Card dataProps={dataProps[0]} />);
    expect(component).toMatchSnapshot();
  });

  it("Should render with multiple data", () => {
    dataProps.forEach((data, index) => {
      let component = renderer.create(<Card dataProps={data} key={index} />);
      expect(component).toMatchSnapshot();
    });
  });
});
