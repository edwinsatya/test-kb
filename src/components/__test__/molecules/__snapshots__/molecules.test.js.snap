// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Should render with dataProps from api 1`] = `
<div>
  <div
    className="undefined w-full"
  >
    <img
      alt="Tribute to Seventeen"
      className="w-full h-20 md:h-48 xl:h-96"
      src="https://kitabisa-userupload-01.s3-ap-southeast-1.amazonaws.com/_production/campaign/97347/31_97347_1545999748_841841_f.png"
    />
  </div>
  <div
    className="flex flex-col p-1 md:p-2 xl:p-3"
  >
    <div
      className="text-[10px] md:text-lg xl:text-2xl font-semibold my-1 xl:my-2 2xl:my-3"
    >
      Tribute to S...
    </div>
    <div
      className="bg-gray-200 h-[4px] md:h-2 xl:h-3 w-full mb-1 xl:mb-2 rounded-md overflow-hidden"
    >
      <div
        className="bg-gray-200 h-[4px] md:h-2 xl:h-3"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="flex flex-col"
    >
      <div
        className="flex justify-between"
      >
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Terkumpul
        </div>
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Sisa hari
        </div>
      </div>
      <div
        className="flex justify-between"
      >
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Rp50.000
        </div>
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Component Should render with multiple data 1`] = `
<div>
  <div
    className="undefined w-full"
  >
    <img
      alt="Tribute to Seventeen"
      className="w-full h-20 md:h-48 xl:h-96"
      src="https://kitabisa-userupload-01.s3-ap-southeast-1.amazonaws.com/_production/campaign/97347/31_97347_1545999748_841841_f.png"
    />
  </div>
  <div
    className="flex flex-col p-1 md:p-2 xl:p-3"
  >
    <div
      className="text-[10px] md:text-lg xl:text-2xl font-semibold my-1 xl:my-2 2xl:my-3"
    >
      Tribute to S...
    </div>
    <div
      className="bg-gray-200 h-[4px] md:h-2 xl:h-3 w-full mb-1 xl:mb-2 rounded-md overflow-hidden"
    >
      <div
        className="bg-gray-200 h-[4px] md:h-2 xl:h-3"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="flex flex-col"
    >
      <div
        className="flex justify-between"
      >
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Terkumpul
        </div>
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Sisa hari
        </div>
      </div>
      <div
        className="flex justify-between"
      >
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Rp50.000
        </div>
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Component Should render with multiple data 2`] = `
<div>
  <div
    className="undefined w-full"
  >
    <img
      alt="Tribute to Seventeen"
      className="w-full h-20 md:h-48 xl:h-96"
      src="https://kitabisa-userupload-01.s3-ap-southeast-1.amazonaws.com/_production/campaign/97347/31_97347_1545999748_841841_f.png"
    />
  </div>
  <div
    className="flex flex-col p-1 md:p-2 xl:p-3"
  >
    <div
      className="text-[10px] md:text-lg xl:text-2xl font-semibold my-1 xl:my-2 2xl:my-3"
    >
      Tribute to S...
    </div>
    <div
      className="bg-gray-200 h-[4px] md:h-2 xl:h-3 w-full mb-1 xl:mb-2 rounded-md overflow-hidden"
    >
      <div
        className="bg-gray-200 h-[4px] md:h-2 xl:h-3"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="flex flex-col"
    >
      <div
        className="flex justify-between"
      >
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Terkumpul
        </div>
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Sisa hari
        </div>
      </div>
      <div
        className="flex justify-between"
      >
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          Rp50.000
        </div>
        <div
          className="text-[6px] md:text-base xl:text-lg font-semibold"
        >
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card Component Should render with no dataProps from api 1`] = `
<div>
  No Data
</div>
`;
